feat: add development docker compose

sergio-toro opened the pull request:
## Description

This PR adds a development environment with all dependencies needed to start `mem0` + `postgres` + `neo4j` graph.


## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (does not change functionality, e.g. code style improvements, linting)
- [x] Documentation update

## How Has This Been Tested?

The development environment spins up when running `docker compose up`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

Dev-Khant commented on mem0/llms/openai.py:
```diff
@@ -1,6 +1,6 @@
 import json
 import os
-from typing import Dict, List, Optional
+from typing import Dict, List, Optional, Union, Any
```
Hey, what do we need to import these?

Dev-Khant commented on mem0/vector_stores/pgvector.py:
```diff
@@ -55,6 +55,8 @@ def __init__(
         self.conn = psycopg2.connect(dbname=dbname, user=user, password=password, host=host, port=port)
         self.cur = self.conn.cursor()
 
+        self.cur.execute("CREATE EXTENSION IF NOT EXISTS vector")
```
You can keep this in `create_col` function.

sergio-toro replied:
modified

Dev-Khant commented on mem0/memory/graph_memory.py:
```diff
@@ -61,7 +61,7 @@ def add(self, data, filters):
         added_entities = self._add_entities(to_be_added, filters["user_id"], entity_type_map)
 
-        return {"deleted_entities": deleted_entities, "added_entities": added_entities}
+        return {"deleted_entities": deleted_entities, "added_entities": added_entities, "relations": to_be_added}
```
What's the reason for adding the `relations` key here?

Dev-Khant commented on mem0/memory/graph_memory.py:
```diff
@@ -344,14 +343,9 @@ def _add_entities(self, to_be_added, user_id, entity_type_map):
                 params = {
                     "source_id": source_node_search_result[0]["elementId(source_candidate)"],
                     "destination_name": destination,
-                    "relationship": relationship,
```
Why are we removing this?

Dev-Khant replied:
Hey TS library is pretty new I think python should be taken as reference here actually :) 
@whysosaket Can you please take a look and confirm here?

sergio-toro replied:
Either way, the relationship parameter is not used on the query, so why adding it?

Dev-Khant reviewed (changes requested):
Can you please also update the docs for it: https://docs.mem0.ai/open-source/features/rest-api?

Dev-Khant reviewed (commented):
Hey @sergio-toro Thanks for addressing the comments. But I think if you are changing the output structure for graph in this PR then you should also update the docs for it: https://docs.mem0.ai/open-source/graph_memory. Or else you can skip the graph changes in this PR.

sergio-toro commented:
> Hey @sergio-toro Thanks for addressing the comments. But I think if you are changing the output structure for graph in this PR then you should also update the docs for it: https://docs.mem0.ai/open-source/graph_memory. Or else you can skip the graph changes in this PR.

Ok, relations have been removed from this PR
